app/managers/performance_metrics_manager.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getTimeSinceStartup} from 'react-native-startup-time';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'channelSwitch' |
+    'teamSwitch';
+
+class PerformanceMetricsManager {
+    private startTimes: {[metricName: string]: undefined | number} = {};
+    private target: Target;
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target) {
+        if (this.target !== location) {
+            return;
+        }
+
+        getTimeSinceStartup().then((v) => console.log(`Took ${v}`));
```

Comment:
How long does this async call take to make? If that takes too long or is inconsistent, it might cause our load times to seem more inconsistent than they should be

Reply:
I haven't looked at the internals, but this should be called and handled "soonish" in the native side. What we are waiting is for the result to come back to the JavaScript side.

Nevertheless, if this takes "long", it is because the system is heavily loaded. A delay here should be more revealing that hurting, right?

Reply:
That's a good point. As long as the delay isn't so big that it drowns out the actual load time or so inconsistent that it makes it hard to benchmark it, that'd be fine then